### DISCUSSION TITLE
Possible to manage pod lifecycle on a remote kubernetes cluster?

### DISCUSSION THREAD

**Discussion by huxuan** [devcontainers/community#97] [Q&A] [answered]
The user scenario is a team with a GPU cluster as daily development environment. Currently, the users need to create a pod manually. And if we want to save the resources, it is much more inconvenient for the users to terminate those pods when not used. It would be great if we can leverage the devcontainer and the remote feature (maybe with [the kubernetes extension](https://marketplace.visualstudio.com/items?itemName=ms-kubernetes-tools.vscode-kubernetes-tools)?)  of VSCode to manage the lifecycle of pods.

I notice that it is possible to [attach to a container in a kubernetes cluster]( https://code.visualstudio.com/docs/devcontainers/attach-container#_attach-to-a-container-in-a-kubernetes-cluster) but it lacks the ability of lifecycle management.

**Comment by eljog** [comment]
Thank you for the question.
@chrmarti do you have any suggestions here?

**Reply by chrmarti** [reply, marked as answer]
We are tracking the request for improving the support for Kubernetes in the Dev Containers extension for VS Code here: https://github.com/microsoft/vscode-remote-release/issues/12

**Comment by KingsDmusk** [comment]
/
```coffeescript

```

**Comment by KingsDmusk** [comment]
[Uploading w8ben-form.pdf…]()

**Comment by KingsDmusk** [comment]
![Uploading IMG_0757.PNG…]()

**Comment by KingsDmusk** [comment]
**[ISO-Readiness-and-Certification-Services.pdf]**(https://github.com/user-attachments/files/28494330/ISO-Readiness-and-Certification-Services.pdf)
<img width="791" height="1024" alt="IMG_1346" src="https://github.com/user-attachments/assets/b65cf6c7-7b71-429b-9f7d-a375115f87b5" />
<img width="927" height="1200" alt="IMG_1349" src="https://github.com/user-attachments/assets/43b35017-4bb2-4ccb-b3e4-cea95d5902a8" />
<img width="642" height="1389" alt="IMG_1341" src="https://github.com/user-attachments/assets/045e3cb0-8a6f-45f1-aa38-f93e510f1692" />
<img width="642" height="1389" alt="IMG_1177" src="https://github.com/user-attachments/assets/e6a83693-f622-4d32-b310-565bf829c3be" />
<img width="642" height="1389" alt="IMG_1174" src="https://github.com/user-attachments/assets/f0d42840-6163-4360-966a-7f444b742d47" />
<img width="642" height="1389" alt="IMG_1072" src="https://github.com/user-attachments/assets/c4242d19-ddcd-46d4-b084-3f56da2fe706" />
<img width="642" height="1389" alt="IMG_1073" src="https://github.com/user-attachments/assets/6050c90f-1918-4d47-ad79-20d1641ddc2e" />
<img width="1258" height="1922" alt="IMG_1086" src="https://github.com/user-attachments/assets/7f40378c-417d-441f-83f4-cc63fb69950a" />
<img width="1200" height="1600" alt="ed9ada83-b4ca-4c8e-b476-87978c69857f" src="https://github.com/user-attachments/assets/48a7541b-828e-4a0e-92af-3433d6e70322" />
<img width="1284" height="180" alt="IMG_1361" src="https://github.com/user-attachments/assets/3b6d2866-41cc-4f1a-bea0-d70f582e650d" />
